### PR TITLE
Update performance measurements macros

### PR DESCRIPTION
**Pull request by mmaka1**
Macros are now configurable with specialized trace code specified by a client. 
Added mcps peak tracing for comp_copy().

### CODE REVIEW

File: src/arch/xtensa/configs/apollolake_defconfig [hunk 1]
```diff
@@ -3,3 +3,4 @@ CONFIG_CAVS_DMIC=y
 CONFIG_CAVS_SSP=y
 CONFIG_LP_MEMORY_BANKS=2
 CONFIG_HP_MEMORY_BANKS=8
+CONFIG_PERFORMANCE_COUNTERS=y
```
Comment (lbetlej) [inline src/arch/xtensa/configs/apollolake_defconfig]: @mmaka1 Any reason this option is added for APL but not for ICL/TGL ?

**Comment by lgirdwood** [comment]
@lbetlej @mmaka1 UTs seem to be failing atm on pipeline start/stop (on most PRs).
```
24: Test command: /localdisk/tools/xtensa/RG-2017.8-linux/XtensaTools/bin/xt-run "--exit_with_target_code" "preproc_varargs_count"
24: Test timeout computed to be: 10000000
6: ok 2 - test_audio_mixer_prepare_no_sources
9: ok 1 - test_audio_pipeline_free_comp_busy
9: *WARNING* Unhandled user exception: StoreProhibitedCause (0x00000038)
9: ( Xtensa ISS ) *WARNING* run exited with status 'virtual breakpoint'
 9/42 Test  #9: pipeline_free ....................***Failed    3.60 sec
```

**Comment by mmaka1** [comment]
@lgirdwood Rebased to master but also had to provide a small fix for UT specifically for this one.

File: src/arch/xtensa/configs/apollolake_defconfig (hunk 1 above)
Comment (mmaka1) [reply src/arch/xtensa/configs/apollolake_defconfig]: I have been observing CNL and APL specifically. If you have another platform to troubleshoot pls turn it on.

**Comment by lgirdwood** [comment]
@mmaka1 there is a internal CI test fail.

File: src/arch/xtensa/configs/apollolake_defconfig (hunk 1 above)
Comment (jajanusz) [reply src/arch/xtensa/configs/apollolake_defconfig]: Shouldn't it be something that is  enabled in  custom-debug build? Doesn't look like config that should be in defconfig.

**Pull request merged**